lib/widgets/current-user.dart

```diff
@@ -12,11 +12,23 @@ class CurrentUser extends StatefulWidget {
 class _CurrentUserState extends State<CurrentUser> {
   final databaseReference = FirebaseDatabase.instance.reference();
   final FirebaseAuth firebaseAuth = FirebaseAuth.instance;
+  FirebaseUser currentUser;
   bool isSwitched = false;
 
   DeviceLocation deviceLocation = DeviceLocation();
   LocationData currentLocation;
 
+  @override
+  initState() {
+    super.initState();
+
+    firebaseAuth.currentUser().then((user) {
```

Comment:
Maybe instead of using the currentUser data, we can use the currentuser.uid to read the user's location node in firebase to get the name/email. 

The benefit also would be that it'll be consistent with where friends of the user are getting their name.